ibis/backends/base/sql/compiler/query_builder.py

```diff
@@ -302,12 +308,28 @@ def format_subqueries(self):
 
         return "WITH {}".format(",\n".join(buf))
 
+    def _fix_name(self, name: str) -> str:
+        return self.translator_class._fix_name(name)
+
     def format_select_set(self):
         # TODO:
         context = self.context
         formatted = []
+        unnests = []
         for node in self.select_set:
-            if isinstance(node, ops.Value):
+            if unnest_nodes := list(an.find_toplevel_unnests(node)):
+                assert (
+                    len(unnest_nodes) == 1
+                ), f"got more than one unnest node: {len(unnest_nodes)}"
+                (unnest_node,) = unnest_nodes
+                expr_str = self._fix_name(unnest_node.name)
+                unnests.extend(map(rels._UnnestTable, unnest_nodes))
+                res = self._translate(node, named=True, permit_subquery=True)
+                # TODO: this is a crime: we're replacing the unnests with their
+                # equivalent *named* expression, and yeah, it's string
+                # replacement
```

Comment:
This todo seems pretty critical - do the names here have uuids (or some other likely-to-be-unique identifier)? Otherwise how likely is a collision here?

Reply:
Changed the approach and got rid of this gunk entirely!